solidity/contracts/converter/types/liquidity-pool-v3/LiquidityPoolV3Converter.sol

```diff
@@ -0,0 +1,1356 @@
+// SPDX-License-Identifier: SEE LICENSE IN LICENSE
+pragma solidity 0.6.12;
+import "../../ConverterVersion.sol";
+import "../../interfaces/IConverter.sol";
+import "../../interfaces/IConverterAnchor.sol";
+import "../../interfaces/IConverterUpgrader.sol";
+import "../../../token/interfaces/IDSToken.sol";
+import "../../../utility/ContractRegistryClient.sol";
+import "../../../utility/ReentrancyGuard.sol";
+import "../../../utility/SafeMath.sol";
+import "../../../utility/TokenHandler.sol";
+import "../../../utility/TokenHolder.sol";
+import "../../../utility/Math.sol";
+
+/**
+  * @dev Liquidity Pool v3 Converter
+  *
+  * The liquidity pool v3 converter is a specialized version of a converter that manages
+  * a classic bancor liquidity pool.
+  *
+  * Even though pools can have many reserves, the standard pool configuration
+  * is 2 reserves with 50%/50% weights.
+*/
+contract LiquidityPoolV3Converter is ConverterVersion, IConverter, TokenHandler, TokenHolder, ContractRegistryClient, ReentrancyGuard {
+    using SafeMath for uint256;
+    using Math for *;
+
+    IERC20Token private constant ETH_RESERVE_ADDRESS = IERC20Token(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE);
+    uint256 private constant MAX_UINT128 = 0xffffffffffffffffffffffffffffffff;
+    uint256 private constant MAX_UINT112 = 0xffffffffffffffffffffffffffff;
+    uint256 private constant AVERAGE_RATE_PERIOD = 10 minutes;
+    uint32 private constant PPM_RESOLUTION = 1000000;
+
+    uint256 public reserveBalances;
```

Comment:
implementation detail - better to have it as an external function that returns 2 uint256 values

Reply:
Done.